Usage of --disable-docs breaks HPC runtime build

**Describe the bug**
I tried to use ```./configure --disable-docs``` when I configured the service build, and then once complete, build the HPC runtime with just ```./configure```. I expected to be able to compile successfully, but instead an error is encountered.

**GEOPM version**
07a8340077e46d7e4f66af42fa86175f828df065

**Expected behavior**
Successful build

**Actual behavior**
Error encountered when make is invoked:
```
make[1]: *** No rule to make target 'man/GEOPM_CXX_MAN_Agent.3', needed by 'all-am'.  Stop.
```

Related to #2215

--- a/service/Makefile.am
+++ b/service/Makefile.am
@@ -335,6 +335,10 @@ rpm: dist
 
 checkprogs: gtest-checkprogs geopmdpy-checkprogs $(check_PROGRAMS) $(check_LTLIBRARIES)
 
+all-local: $(all_local)
+
+all_local =
+
 CLEAN_LOCAL_TARGETS = clean-local-python
 
 clean-local: $(CLEAN_LOCAL_TARGETS)

--- a/service/docs/Makefile.mk
+++ b/service/docs/Makefile.mk
@@ -195,14 +195,13 @@ base_man = docs/build/man/geopmadmin.1 \
            docs/build/man/geopmpy.7 \
            # end
 
+all_man_target = docs/build/man/.dirfile
 all_man = $(dist_man_MANS) $(base_man)
+all_local += $(all_man_target)
 
 docs: docs_man docs_html
 
 if ENABLE_DOCS
-
-all_man_target = docs/build/man/.dirfile
-
 $(all_man_target): $(all_man_rst)
 	LD_LIBRARY_PATH=.libs:$(LD_LIBRARY_PATH) \
 	PYTHONPATH=$(abs_srcdir):$(PYTHONPATH) \
@@ -223,6 +222,9 @@ docs_man: libgeopmd.la $(abs_srcdir)/geopmdpy/version.py
 	sphinx-build -M man $(abs_srcdir)/docs/source docs/build
 
 else
+
+$(all_man_target): $(all_man)
+	touch $(all_man_target)
 
 $(all_man): docs/build/man/%: $(top_srcdir)/docs/source/%.rst
 	mkdir -p docs/build/man

